docs/source/markdown/podman-attach.1.md

```diff
@@ -9,48 +9,52 @@ podman\-attach - Attach to a running container
 **podman container attach** [*options*] *container*
 
 ## DESCRIPTION
-The attach command allows you to attach to a running container using the container's ID
-or name, either to view its ongoing output or to control it interactively.
+**podman attach** allows you to attach to a running *container* using the *container's name* or *ID*, either to view its ongoing output or to control it interactively.
 
-You can detach from the container (and leave it running) using a configurable key sequence. The default
+You can detach from the *container* (and leave it running) using a configurable key sequence. The default
 sequence is `ctrl-p,ctrl-q`.
 Configure the keys sequence using the **--detach-keys** option, or specifying
-it in the **containers.conf** file: see **containers.conf(5)** for more information.
+it in the `containers.conf` file: see [**containers.conf(5)**](https://github.com/containers/common/blob/master/docs/containers.conf.5.md) for more information.
 
 ## OPTIONS
-#### **--detach-keys**=*sequence*
+#### **--detach-keys**=**sequence**
 
-Specify the key sequence for detaching a container. Format is a single character `[a-Z]` or one or more `ctrl-<value>` characters where `<value>` is one of: `a-z`, `@`, `^`, `[`, `,` or `_`. Specifying "" will disable this feature. The default is *ctrl-p,ctrl-q*.
+Specify the key **sequence** for detaching a *container*. Format is a single character `[a-Z]` or one or more `ctrl-<value>` characters where `<value>` is one of: `a-z`, `@`, `^`, `[`, `,` or `_`. Specifying "" will disable this feature. The default is `ctrl-p,ctrl-q`.
 
-#### **--latest**, **-l**
+#### **--latest**, **-l** [=**false** | *true*]
 
-Instead of providing the container name or ID, use the last created container. If you use methods other than Podman
-to run containers such as CRI-O, the last started container could be from either of those methods. (This option is not available with the remote Podman client)
+Instead of providing the *container name* or *ID*, use the last created *container*. If you use methods other than Podman to run containers such as CRI-O, the last started *container* could be from either of those methods. The default is **false**.\
+*IMPORTANT: This option is not available with the remote Podman client*
 
-#### **--no-stdin**
+#### **--no-stdin** [=**false** | *true*]
 
-Do not attach STDIN. The default is false.
+Do not attach STDIN. The default is **false**.
```

Comment:
This isn't your change, but this is one of those cases that is nearly impossible for a human brain to read unambiguously. First, I don't believe that the default is false -- that makes no sense. Second, it really needs clarification. I think this should be something like "--no-stdin [=true|false" .... By default, **podman attach** includes stdin." But I think this needs clarification from other team members, because in my testing I can't actually get stdin to work in podman attach no matter what I do.

Reply:
I do believe that the default is false. Because it is --**no**-stdin and not just --stdin. https://github.com/containers/podman/blob/52dae693da0df1447b7f5210a4c842d5c5a8a401/cmd/podman/containers/attach.go#L52

I thought about how to express it alot, so I came up with this definition that is in the MANPAGE_SYNTAX:
If the OPTION is boolean, it is not requiered to set an argument. In this case the equal sign is put in brackets `[]` with the arguments, in difference to arguments that are not boolean as shown with **--answer**. The default argument that is shown for a boolean OPTION is given when the OPTION is not used in the command. If the OPTION is used in the command and no argument is specified, the OPTION will always become `true`.

Reply:
Sorry, I wasn't clear. The ambiguity is in the meaning of "default": A reader of the man page, seeing the following documentation string:
```
    **--no-stdin** [=**false** | *true*]    [...] The default is **false**
```
...might have trouble understanding exactly how the flag operates:
* Does "default" apply to **the flag**? That is, if I, the end  user, specify `--no-stdin`, is that the same as `--no-stdin=false`, and hence, I need to explicitly specify `--no-stdin=true` to get the desired effect? Or,
* Does "default" apply to **podman itself**? That is, is the man page saying that `no-stdin` is false by default, hence podman does attach stdin? 
* 
Double negatives are evil. This is a horrible situation that needs clarification. IMO I would completely remove the `[=**false** | *true*]` string from the flag description, and remove "The default is false". They add clutter and confusion. Less is more:
```
   #### **--no-stdin**
   Do not attach STDIN.
```

Reply:
> Does "default" apply to podman itself? That is, is the man page saying that no-stdin is false by default, hence podman does attach stdin?

Yes this is what I mean. In the end it is optinial if you want to put an argument behind the option. This is why the whole thing is in `[]` including the equal sign. But since the default value of boolean options is sometimes `true` and sometimes `false` I thought it needed clarification. I had the choice to define what is default for me. 

> Does "default" apply to the flag? That is, if I, the end user, specify --no-stdin, is that the same as --no-stdin=false

No, thats is why I said in the MANPAGE_SYNTAX that if you type a option without putting anything behind it, it is always true. I believe this is the way in which it is currently working. 

> Double negatives are evil. This is a horrible situation that needs clarification.

That was my starting point and I think that to. But it not on me to change the option itself but in this case I would like to have it changed. 

>I would completely remove the [=**false** | *true*] string from the flag description, and remove "The default is false". They add clutter and confusion.

I though since there are these different cases of options that sometimes default to `true` and sometimes to `false` it should be mentioned. So you know what option will be in command automaticly if you dont specifiy them.

Reply:
>No, thats is why I said in the MANPAGE_SYNTAX that if you type a option without putting anything behind it, it is always true. 

I interpret that as "Please read a long complicated document before you are allowed to understand our man pages".  I don't think that's fair to end users, who are simply trying to solve a problem by finding one small relevant piece in a large documentation set. I feel pretty strongly about removing the `[=false]` thing from the description, I think if you look at it from the perspective of a normal user, one without your background and expertise, you will agree that it is too confusing.

Reply:
You are completely right. For the normal user it is very confusing. I will remove it.

Just in general, is there no where this fits. Cause the only way currently for users to find out about it, is if they look into the code itself. I don't have a problem with that, but I have the feeling it should be mentioned somewhere. But you are right, maybe not in the manpages.

Reply:
@edsantiago @TomSweeneyRedHat @rhatdan @vrothberg 
What is the general opinion about the topic above. In the end, I find it really important. So it should be mentioned somewhere. I could live with leaving out the arguments `false` and `true` in the option line. But I would at least like to specifiy the default value in the description with a short sentence.